modules tcl: simplify env modification block in template

xdelaruelle opened the pull request:
Simplify environment modification block in modulefile Tcl template by always setting a path delimiter to the `prepend-path`, `append-path` and `remove-path` commands.

Remove `--delim` option to the setenv command as this command does not allow such option.

Add test.

xdelaruelle committed:
modules tcl: simplify env modification block in template

Simplify environment modification block in modulefile Tcl template by
always setting a path delimiter to the prepend-path, append-path and
remove-path commands.

Remove --delim option to the setenv command as this command does not
allow such option.

Add test.

alalazo commented:
Do we also need to bump the Module file version at the top of the template?

xdelaruelle commented:
> Do we also need to bump the Module file version at the top of the template?

No, the `--delim` option is also supported on old Modules version (v3.2).

xdelaruelle commented:
If this one is ok, I will look at the `append_flag`, `remove_flag` support to finish what was started in #19577

alalazo commented on lib/spack/spack/test/modules/tcl.py:
```diff
@@ -108,6 +108,18 @@ def test_alter_environment(self, modulefile_content, module_configuration, confi
+        module_configuration("module_path_separator")
+        content = modulefile_content("module-path-separator")
+
+        for line in content:
+            if re.match(r"[a-z]+-path .* COLON ", line):
+                assert ' --delim ":" ' in line
+            elif re.match(r"[a-z]+-path .* SEMICOLON ", line):
+                assert ' --delim ";" ' in line
```
Code looks good to me, but I think an empty module file would pass this test?

xdelaruelle replied:
You are right. I tried to copy/paste the same test used for lmod, but I think it is better to change the logic and explicitly test the explicit output like done on `test_alter_environment`

xdelaruelle committed:
modules lmod: fix test_prepend_path_separator test

Update test_prepend_path_separator test to explicitly check the 6
path-like commands that should be present in generated modulefile.

Also fix a small typo on test_alter_environment test.

alalazo commented on lib/spack/spack/test/modules/tcl.py:
```diff
@@ -108,6 +108,19 @@ def test_alter_environment(self, modulefile_content, module_configuration, confi
         assert len([x for x in content if "setenv LIBDWARF_ROOT" in x]) == 1
 
+    def test_prepend_path_separator(self, modulefile_content, module_configuration):
+        """Tests modifications to run-time environment."""
```
```suggestion
        """Tests that we can use custom delimiters to manipulate path lists."""
```

xdelaruelle replied:
@alalazo Commits have been updated to apply your suggestion.

alalazo reviewed (dismissed):
One minor comment, otherwise LGTM

alalazo reviewed (approved):
Thanks! You also improved a competitor unit test so :bow: :laughing:
